src/metabase/automagic_dashboards/rules.clj

```diff
@@ -300,14 +301,20 @@
                              e)))
       nil)))
 
+(defn- trim-trailing-slash
+  [s]
+  (if (str/ends-with? s "/")
```

Comment:
Are you sure `s` will always have a value here? `str/ends-with?` will throw an NPE.

Reply:
The way it's currently used, yes (only called on `getFileName` of a `Path` that `isDirectory`).

Reply:
Part of the changes in this PR is also, that we always throw rather than return nil.